test/https-with-ca-option.js

```diff
@@ -30,15 +30,32 @@ t.before(() => new Promise((res) => {
 }))
 t.teardown(() => server.close())
 
+const failure = {
+  name: 'FetchError',
+  message: `request to ${base}hello failed, reason: unable to verify the first certificate`,
+  code: 'UNABLE_TO_VERIFY_LEAF_SIGNATURE',
+  errno: 'UNABLE_TO_VERIFY_LEAF_SIGNATURE',
+  type: 'system',
+}
+
 // this test will fail after Jan 30 23:23:26 2025 GMT
 t.test('make https request without ca, should fail', t =>
-  t.rejects(fetch(`${base}hello`), {
-    name: 'FetchError',
-    message: `request to ${base}hello failed, reason: unable to verify the first certificate`,
-    code: 'UNABLE_TO_VERIFY_LEAF_SIGNATURE',
-    errno: 'UNABLE_TO_VERIFY_LEAF_SIGNATURE',
-    type: 'system',
-  }))
+  t.rejects(fetch(`${base}hello`), failure))
+
+t.test('make https request with NODE_TLS_REJECT_UNAUTHORIZED set to 1, should fail', async t => {
+  process.env.NODE_TLS_REJECT_UNAUTHORIZED = '1'
+  t.rejects(fetch(`${base}hello`), failure)
+  delete process.env.NODE_TLS_REJECT_UNAUTHORIZED
```

Comment:
```suggestion
  process.env.NODE_TLS_REJECT_UNAUTHORIZED = '1'
  t.teardown(() => delete process.env.NODE_TLS_REJECT_UNAUTHORIZED)
  await t.rejects(fetch(`${base}hello`), failure)
```

No functional difference, but I like to keep the habit of `await` any `t.rejects` calls in case more assertions are added in the future.

Also it's helpful to use `t.teardown()` to undo any `process.env` setting since it will run in case the whole `t.test` errors out and won't leak into any other tests.